package-drone/src/main/java/de/dentrassi/pm/jenkins/DroneRecorder.java

```diff
@@ -485,19 +433,40 @@ public Void invoke ( final File basedir, final VirtualChannel channel ) throws I
                             filename = f;
                         }
                         uploader.upload ( file, filename );
-//                        TODO this.artifacts.put ( artifacId, f );
                     }
-
+                    uploadResult.addArtifacts ( uploader.complete () );
```

Comment:
Upload is called for each individual file and does what is necessary for the files. Complete finalized the transaction (maybe does the actual upload (like V3)). It returns void and throws IOExceptions. For V2 it does nothing (I should do an empty implementation in AbstractUploader probably).
getUploadedArtifacts just returns the Map and throws no Exceptions